src/http.c

```diff
@@ -336,8 +336,23 @@ http_filename(const http_t *conn, char *filename)
 {
 	const char *h;
 	if ((h = http_header(conn, "Content-Disposition:")) != NULL) {
-		sscanf(h, "%*s%*[ \t]filename%*[ \t=\"\'-]%254[^\n\"\' ]",
+		/**
+		 * This header has these formats:
+		 * Content-Disposition: inline
+		 * Content-Disposition: attachment
+		 * Content-Disposition: attachment; filename="filename.jpg"
+		 */
+		sscanf(h, "%*s%*[ \t]filename%*[ \t=\"\'-]%254[^\n\"\']",
 		       filename);
+		if (filename == 0)
+			return;
+		/* Trim spaces at the end of string */
+		for (i = strlen(filename) - 1; i > 0; i--) {
```

Comment:
Instead of using `strlen` here, you could (just like the code below it) use `strpbrk`  to find the character to reject,
and `strspn` to skip a run, until you find a zero after the run.